livekit-rtc/livekit/rtc/participant.py

```diff
@@ -539,6 +549,119 @@ async def set_attributes(self, attributes: dict[str, str]) -> None:
         finally:
             FfiClient.instance.queue.unsubscribe(queue)
 
+    async def stream_text(
+        self,
+        *,
+        destination_identities: List[str] = [],
+        topic: str = "",
+        extensions: Dict[str, str] = {},
+        reply_to_id: str | None = None,
+        total_size: int | None = None,
+    ) -> TextStreamWriter:
+        """
+        Returns a TextStreamWriter that allows to write individual chunks of text to a text stream.
+        In most cases where you want to simply send a text message use send_text() instead.
+        """
+        writer = TextStreamWriter(
+            self,
+            topic=topic,
+            extensions=extensions,
+            reply_to_id=reply_to_id,
+            destination_identities=destination_identities,
+            total_size=total_size,
+        )
+
+        await writer._send_header()
+
+        return writer
+
+    async def send_text(
+        self,
+        text: str,
+        *,
+        destination_identities: List[str] = [],
+        topic: str = "",
+        extensions: Dict[str, str] = {},
+        reply_to_id: str | None = None,
+    ):
+        total_size = len(text.encode())
+        writer = await self.stream_text(
+            destination_identities=destination_identities,
+            topic=topic,
+            extensions=extensions,
+            reply_to_id=reply_to_id,
+            total_size=total_size,
+        )
+
+        for chunk in split_utf8(text, STREAM_CHUNK_SIZE):
+            await writer.write(chunk)
+        await writer.aclose()
+
+        return writer.info
+
+    async def stream_bytes(
+        self,
+        name: str,
+        *,
+        total_size: int | None = None,
+        mime_type: str = "application/octet-stream",
+        extensions: Optional[Dict[str, str]] = None,
```

Comment:
Let's do like the send_text method
`extensions: dict[str, str]`

Reply:
I specifically changed this to optional because @longcw had concerns about defaulting a list to an empty list in method parameters

Reply:
see https://github.com/livekit/python-sdks/pull/347#discussion_r1920930040